wavecrestweb/kenworthy-machine
# #KM-40: Connect footer to TinaCMS

## Description

This PR connects footer to TinaCMS. After updating the TinaCMS config to include footer collection schema, a custom query was created to fetch footer data along with the page data. The Footer component was removed from the layout file and added to every page instead. Lastly, the Footer component was refactored to replace static imports and hardcoded data with data from TinaCMS.

## Related Tickets & Documents
Closes #KM-40

## Acceptance Criteria

- [x] Footer fetches all it's data from TinaCMS
- [x] Footer's content is editable via TinaCMS

## What type of PR is this? (check all applicable)


|     | Type                       |
| --- | -------------------------- |
|  `✓`   | :sparkles: New feature     |
| `✓`    | :hammer: Refactoring       |


## Mobile & Desktop Screenshots/Recordings
<img width="1264" alt="Screen Shot 2024-07-23 at 2 33 50 PM" src="https://github.com/user-attachments/assets/4c575671-c978-4be5-94fb-2a448d70d4e0">

## What I learned

You can create custom queries in TinaCMS to query 2 or more different collections at a time instead of querying them separately. This can be achieved by creating a `tina/queries/main.gql` file which will contain custom queries. As with other automatically generated queries, the types will be automatically generated for custom queries as well. 

The `global` property set to `true` in the TinaCMS collection's ui allows editing of the collection from any page. The global collections can be accessed through the sidebar in TinaCMS visual editor.

## Testing steps

1. `git switch KM-40/refactor-footer`
2. `npm run dev`
3. Open http://localhost:3000/ and check that footer is displayed properly on all the pages. 
4. Check that the footer navigation links work as expected. 
5. Open TinaCMS editor and try to change footer's content.

## Added to documentation?

|     | Type                       |
| --- | -------------------------- |
|     | 📜 README.md    |
|  `✓`  | 🙅 no documentation needed    |


## What gif best describes this PR?
  ![gif name](https://giphy.com/gifs/iFkHQLzYA09Zm/giphy.gif)
